File: scala/dao/interfaces/MealInterface.scala
package dao.interfaces

import entities.Meal
trait MealInterface {
  def findAllByUserId(id: Long): Seq[Meal]
  def findAllTopical(id: Long): Seq[Meal]
  def deleteMeal(id: Long): Boolean
  def addMeal(v: Meal): Boolean
  def updateMeal(id: Long, amount: Double): Boolean
}

File: lab3/src/main/java/dao/interfaces/MealInterface.java
package dao.interfaces;

import entities.Meal;
import java.util.List;

public interface MealInterface {
    List<Meal> findAllByUserId(Long id);
    List<Meal> findAllTopical(Long id);
    boolean deleteMeal(Long id);
    boolean addMeal(Meal v);
    boolean updateMeal(Long id, Double amount);
}
